src/lib/CMakeLists.txt

```diff
@@ -0,0 +1,184 @@
+    ${LIB_DIR}/common/OrderedKVPLineParser.cpp
+    ${LIB_DIR}/common/OrderedSeparatorLineParser.cpp
+    ${LIB_DIR}/common/lpm_u5.c
+    ${LIB_DIR}/common/lptimTick_u5.c
+    ${LIB_DIR}/common/nvmPartition.cpp
+    ${LIB_DIR}/common/pcap.c
+    ${LIB_DIR}/common/trace.c
+    ${LIB_DIR}/common/reset_reason.c
+    ${LIB_DIR}/common/serial.c
+    ${LIB_DIR}/common/sensorWatchdog.cpp
+    ${LIB_DIR}/common/serial_console_u5.cpp
+    ${LIB_DIR}/common/stress.c
+    ${LIB_DIR}/common/tokenize.c
+    ${LIB_DIR}/common/uptime.c
+    ${LIB_DIR}/common/util.c
+    ${LIB_DIR}/common/watchdog.c
+    ${LIB_DIR}/sys/ram_partitions.c
+    ${LIB_DIR}/sys/bm_config_wrapper.cpp
+    ${LIB_DIR}/debug/debug.c
+    ${LIB_DIR}/debug/debug_spotter.cpp
+    ${LIB_DIR}/debug/debug_gpio.c
+    ${LIB_DIR}/debug/debug_htu.cpp
+    ${LIB_DIR}/debug/debug_i2c.c
+    ${LIB_DIR}/debug/debug_spi.c
+    ${LIB_DIR}/debug/debug_pluart_cli.cpp
+    ${LIB_DIR}/debug/debug_pressure_sensor.cpp
+    ${LIB_DIR}/debug/debug_memfault.c
+    ${LIB_DIR}/debug/debug_rtc.c
+    ${LIB_DIR}/debug/debug_ina232.cpp
+    ${LIB_DIR}/debug/debug_sys.cpp
+    ${LIB_DIR}/debug/debug_w25.cpp
+    ${LIB_DIR}/debug/debug_nvm_cli.cpp
+    ${LIB_DIR}/debug/debug_dfu.cpp
+    ${LIB_DIR}/debug/debug_bm_service.cpp
+    ${LIB_DIR}/debug/debug_configuration.cpp
+    ${LIB_DIR}/debug/debug_tca9546a.cpp
+    ${LIB_DIR}/drivers/abstract/abstract_i2c.cpp
+    ${LIB_DIR}/drivers/abstract/abstract_spi.cpp
+    ${LIB_DIR}/drivers/bm_dfu_wrapper.cpp
+    ${LIB_DIR}/drivers/bm_rtc_wrapper.c
+    ${LIB_DIR}/drivers/adin2111/src/adi_hal.c
+    ${LIB_DIR}/drivers/bme280driver.cpp
+    ${LIB_DIR}/drivers/htu21d.cpp
+    ${LIB_DIR}/drivers/ina232.cpp
+    ${LIB_DIR}/drivers/nau7802.cpp
+    ${LIB_DIR}/drivers/ms5803.cpp
+    ${LIB_DIR}/drivers/pca9535.c
+    ${LIB_DIR}/drivers/protected/protected_i2c.c
+    ${LIB_DIR}/drivers/protected/protected_spi.c
+    ${LIB_DIR}/drivers/stm32_adc.c
+    ${LIB_DIR}/drivers/stm32_io.c
+    ${LIB_DIR}/drivers/stm32_rtc.c
+    ${LIB_DIR}/drivers/TSYS01.cpp
+    ${LIB_DIR}/drivers/w25.cpp
+    ${LIB_DIR}/drivers/tca9546a.cpp
+    ${LIB_DIR}/lwip/lwip_support.c
+    ${LIB_DIR}/mcuboot/mcuboot_cli.c
+    ${LIB_DIR}/mcuboot/port_flash.c
+    ${LIB_DIR}/mcuboot/port_misc.c
+    ${LIB_DIR}/memfault/memfault_platform_core_u5.c
+    ${LIB_DIR}/sensor_sampler/sensorSampler.cpp
+    ${LIB_DIR}/sensor_sampler/powerSampler.cpp
+    ${LIB_DIR}/sensor_sampler/htuSampler.cpp
+    ${LIB_DIR}/sensor_sampler/loadCellSampler.cpp
+    ${LIB_DIR}/sensor_sampler/pressureSampler.cpp
+    ${TINYUSB_FILES}
+    ${BM_SERIAL_FILES}
+    ${BM_INTEGRATION_FILES}
+)
+
+if(APP_NAME MATCHES "bridge" AND
+   NOT APP_NAME MATCHES "bringup")
+    list(APPEND LIB_FILES
+        ${LIB_DIR}/debug/debug_bridge_power_controller.cpp
+        ${BM_NCP_FILES}
+    )
+endif()
+
+if(APP_NAME MATCHES "bringup")
+    list(APPEND LIB_FILES
+        ${LIB_DIR}/debug/debug_uart.c
+        ${LIB_DIR}/debug/debug_adin_raw.c
+    )
+endif()
```

Comment:
Haha! Just highlighting as an interesting hack. I see you moved some files into the bridge app folder, but I guess these generally useful, so should stay in lib, even if they're only used in a couple apps. 👏

Reply:
Ah! The BMDK one below is similar too.